LykkeApi2/Controllers/AffiliateController.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Common.Log;
+using Core.Constants;
+using Lykke.Service.Affiliate.Client;
+using Lykke.Service.Affiliate.Contracts;
+using Lykke.Service.RateCalculator.Client;
+using Lykke.Service.RateCalculator.Client.AutorestClient.Models;
+using LykkeApi2.Infrastructure;
+using LykkeApi2.Models.Affiliate;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Rest;
+
+namespace LykkeApi2.Controllers
+{
+    [Authorize]
+    [Route("api/[controller]")]
+    [Produces("application/json")]
+    public class AffiliateController : Controller
+    {
+        private readonly IAffiliateClient _affiliateClient;
+        private readonly IRequestContext _requestContext;
+        private readonly IRateCalculatorClient _rateCalculatorClient;
+        private readonly ILog _log;
+
+        public AffiliateController(IAffiliateClient affiliateClient, IRequestContext requestContext, IRateCalculatorClient rateCalculatorClient, ILog log)
+        {
+            _affiliateClient = affiliateClient;
+            _requestContext = requestContext;
+            _rateCalculatorClient = rateCalculatorClient;
+            _log = log;
+        }
+
+        [HttpGet]
+        [Route("link")]
+        [ProducesResponseType(typeof(AffiliateLinkResponse), (int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetLink()
+        {
+            var links = await _affiliateClient.GetLinks(_requestContext.ClientId);
+            var link = links.FirstOrDefault();
+
+            if (link != null)
+            {
+                return Ok(new AffiliateLinkResponse
+                {
+                    Url = link.Url,
+                    RedirectUrl = link.RedirectUrl
+                });
+            }
+            else
+            {
+                return NotFound();
+            }
+        }
+
+        [HttpPost]
+        [Route("create")]
+        [ProducesResponseType(typeof(AffiliateLinkResponse), (int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> CreateLink()
+        {
+            LinkModel link;
+            try
+            {
+                const string redirectUrl = "https://lykke.com";
+                link = await _affiliateClient.RegisterLink(_requestContext.ClientId, redirectUrl);
+            }
+            catch (HttpOperationException e)
+            {
+                await _log.WriteErrorAsync(nameof(AffiliateController), nameof(GetStats), e);
+                return BadRequest(new { message = e.Response.Content });
```

Comment:
Why BadRequest? As i understand, HttpOperationException may occurs only on service down or 500, it's not bad request. In this case, it looks like request is always good, but if problem has occured we need to set 500 as a response code.